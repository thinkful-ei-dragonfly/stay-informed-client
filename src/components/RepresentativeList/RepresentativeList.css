#rep-pane {

}

.headshot {
  /* display: relative; */
  /* float: left; */
  width: 200px;
  height: auto;
}

ul {
  list-style-type: none;
}

/* 
.go-details {
  margin-top: 20px;
  width: 250px;
  padding-left: 10px;
  padding-right: 10px;
  background-color: rgba(221, 221, 221, .5);
  border: 1px solid gray;
  color: black;
  text-align: center;
  font-size: 18px;
  font-weight: bold;
  margin: 4px 2px;
  transition: 0.3s;
  cursor: pointer;
  border-radius: 5%;
} */

/* .go-details:hover {
  background-color: rgba(97, 163, 204, 0.4);
} */
