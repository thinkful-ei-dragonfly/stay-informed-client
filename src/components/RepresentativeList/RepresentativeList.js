import React from 'react';
import './RepresentativeList.css';
import UserContext from '../../contexts/UserContext';

export default class RepresentativeList extends React.Component {
  static contextType = UserContext;

  // This generates the element for each representative tile in the pane
  generateRepList() {
    if (this.context.representatives) {
      const repElems = this.context.representatives.map((rep, idx) => {
        const title = rep.roles[0].short_title;
        // get district if representative
        const district =
          title === 'Rep.' ? `District ${rep.roles[0].district}` : '';
        // to check if photo exists for representative
        const photoUrl = rep.photoUrl;
        const party = (rep.roles[0].party === 'R') ? 'Republican' : 'Democrat';
        return (
          <li key={idx} className="representative">
            {photoUrl ? (
              <img
                className="headshot"
                src={rep.photoUrl}
                alt={`professional photographic headshot of ${rep.first_name} ${
                  rep.last_name
                }`}
              />
            ) : (
              <img
                className="headshot"
                src={'http://media2.giphy.com/gifsu/5zkNlgUl7QodTlg7nT/giphy-glitter.gif'}
                alt={`professional photographic headshot of ${rep.first_name} ${
                  rep.last_name
                }`}
              />
            )}
            <h3 className="rep-name">{`${title} ${rep.first_name} ${
              rep.last_name
            }`}</h3>
            <p>
              <span className="rep-field">State: </span>
              <span className="field-val">
                {rep.roles[0].state} {district}
              </span>
            </p>
            <p>
              <span className="rep-field">Party: </span>
              <span className="field-val">{party}</span>
            </p>
            <button
              onClick={e => this.props.handleClickRepDetails(e, rep.member_id)}
              className="go-details submit"
            >
              Learn More
            </button>
          </li>
        );
      });
      return repElems;
    }
  }

  render() {
    const repElemList = this.generateRepList();

    return (
      <aside className="rep-pane">
        <h2>Your Congress Representatives</h2>
        <ul id="rep-list">{repElemList}</ul>
      </aside>
    );
  }
}
