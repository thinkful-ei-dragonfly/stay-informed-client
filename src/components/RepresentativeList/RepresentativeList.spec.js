import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RepresentativeList from './RepresentativeList'

describe(`RepresentativeList component`, () => {
  it('renders a form.RepresentativeList by default', () => {
    const wrapper = shallow(<RepresentativeList />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
