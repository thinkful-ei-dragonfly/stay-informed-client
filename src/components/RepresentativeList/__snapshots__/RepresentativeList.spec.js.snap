// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepresentativeList component renders a form.RepresentativeList by default 1`] = `
<div
  id="rep-pane"
>
  <h2>
    Your Congress Representatives
  </h2>
  <ul
    id="rep-list"
  />
</div>
`;
